src/CodeGen_Internal.cpp

```diff
@@ -388,6 +388,24 @@ Expr lower_int_uint_mod(const Expr &a, const Expr &b) {
     }
 }
 
+Expr ulong_div(Expr num, Expr den) {
+    internal_assert(num.type() == den.type());
+    Type ty = num.type();
+    internal_assert(ty.is_uint());
+    const int times = ty.bits();
```

Comment:
I think this algorithm should have a few comments explaining it.

Reply:
Right. I'll add some comments.